cirq-core/cirq/devices/noise_utils.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Sequence
+import warnings
+import numpy as np
+
+if TYPE_CHECKING:
+    import cirq
+
+
+# Tag for gates to which noise must be applied.
+PHYSICAL_GATE_TAG = 'physical_gate'
+
+
+@dataclass(frozen=True)
+class OpIdentifier:
+    """Identifies an operation by gate and (optionally) target qubits."""
+
+    gate_type: type
+    qubits: Sequence['cirq.Qid']
+
+    def __init__(self, gate_type: type, *qubits: 'cirq.Qid'):
+        object.__setattr__(self, 'gate_type', gate_type)
+        object.__setattr__(self, 'qubits', qubits)
+
+    def swapped(self):
+        return OpIdentifier(self.gate_type, *self.qubits[::-1])
```

Comment:
is this method necessary? My personal preference is to limit the API as much as possible to reduce complexity.

Reply:
To be resolved by `cirq.Gateset` change.

Reply:
Removed.